### ISSUE TITLE
IMAP will stop working with Google accounts from May 30, 2022

### ISSUE THREAD

**Issue by sepen** [gutmensch/docker-dmarc-report#18]
Hi, thanks for this great project.

Starting May 30, 2022, ​​Google will no longer support the use of third-party apps or devices which ask you to sign in to your Google Account using only your username and password.
https://support.google.com/accounts/answer/6010255?hl=en&visit_id=637891856465928018-837798669&p=less-secure-apps&rd=1

Last version of IMAPClient is used and I don't see too much movement https://github.com/plobbes/mail-imapclient/issues

The generated error is something like this:
`Using Mail::IMAPClient version 3.43 on perl 5.032000
Connecting with IO::Socket::SSL PeerAddr imap.gmail.com PeerPort 993 Proto tcp Timeout 600 Debug 1 SSL_verify_mode 0
Connected to imap.gmail.com
Read: 	* OK Gimap ready for requests from 234.234.234.234 e3mb266103497wrd
Connecting with IO::Socket::SSL PeerAddr imap.gmail.com PeerPort 993 Proto tcp Timeout 600 Debug 1 SSL_verify_mode 0
Connected to imap.gmail.com
Read: 	* OK Gimap ready for requests from 234.234.234.234 g21mb63565606wmh
Sending: 1 LOGIN this-is-my-email-account@gmail.com [Redact: Count=1 Showcredentials=OFF]
Sent 56 bytes
Read: 	1 NO [AUTHENTICATIONFAILED] Invalid credentials (Failure)
ERROR: 1 NO [AUTHENTICATIONFAILED] Invalid credentials (Failure) at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 1388.
	Mail::IMAPClient::__ANON__("1 NO [AUTHENTICATIONFAILED] Invalid credentials (Failure)\x{d}\x{a}") called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 1424
	Mail::IMAPClient::_get_response(Mail::IMAPClient=HASH(0x7fddbd38c350), 1, undef) called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 1350
	Mail::IMAPClient::_imap_command_do(Mail::IMAPClient=HASH(0x7fddbd38c350), "LOGIN this-is-my-email-account\@gmail.com G-b-K-AMlR8Q2nyltB") called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 1248
	Mail::IMAPClient::_imap_command(Mail::IMAPClient=HASH(0x7fddbd38c350), "LOGIN this-is-my-email-account\@gmail.com G-b-K-AMlR8Q2nyltB") called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 601
	Mail::IMAPClient::login(Mail::IMAPClient=HASH(0x7fddbd38c350)) called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 424
	Mail::IMAPClient::Socket(Mail::IMAPClient=HASH(0x7fddbd38c350), IO::Socket::SSL=GLOB(0x7fddbd417588)) called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 369
	Mail::IMAPClient::connect(Mail::IMAPClient=HASH(0x7fddbd38c350)) called at /usr/bin/dmarcts-report-parser.pl line 309
ERROR: 1 NO [AUTHENTICATIONFAILED] Invalid credentials (Failure) at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 1298.
	Mail::IMAPClient::_imap_command(Mail::IMAPClient=HASH(0x7fddbd38c350), "LOGIN this-is-my-email-account\@gmail.com G-b-K-AMlR8Q2nyltB") called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 601
	Mail::IMAPClient::login(Mail::IMAPClient=HASH(0x7fddbd38c350)) called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 424
	Mail::IMAPClient::Socket(Mail::IMAPClient=HASH(0x7fddbd38c350), IO::Socket::SSL=GLOB(0x7fddbd417588)) called at /usr/local/share/perl5/site_perl/Mail/IMAPClient.pm line 369
	Mail::IMAPClient::connect(Mail::IMAPClient=HASH(0x7fddbd38c350)) called at /usr/bin/dmarcts-report-parser.pl line 309
Sending: 2 STATUS Inbox (MESSAGES)`

**Comment by ptman** [comment]
Haven't some people reported success using a gmail account protected by 2FA and then creating an app password?

**Comment by gene1wood** [comment]
Ya, can't you just use an app password? https://support.google.com/accounts/answer/185833

**Comment by landcraft** [comment]
app passwords are no longer supported. any other alternative email provider you suggest?

**Comment by gene1wood** [comment]
@landcraft Can you say more about what you mean by not supported? When I follow [the instructions on Google](https://support.google.com/accounts/answer/185833), I'm able to create an app password :

![Creating an app password on Google 2022-10-05](https://user-images.githubusercontent.com/1134034/194087881-5b2bb9b7-483e-4236-b4cf-45c138fdfd14.png)

**Comment by landcraft** [comment]
See here - https://support.google.com/accounts/answer/6010255

**Comment by gene1wood** [comment]
@landcraft On that page it says

> from May 30, 2022, ​​Google no longer supports the use of third-party apps or devices which ask you to sign in to your Google Account using only your **username and password**.

You can still use app passwords (which are different from you username and password)

Down at the bottom of that page it says to use app passwords and links to [the page that describes how](https://support.google.com/accounts/answer/185833)

![Use app passwords](https://user-images.githubusercontent.com/1134034/194093632-92307d54-ab5f-471d-9005-3b3cfbd1feff.png)

Have you tried provisioning an app password [per those instructions](https://support.google.com/accounts/answer/185833) and using it in `docker-dmarc-report`?

**Comment by landcraft** [comment]
you're right - I can create an app password (didn't have 2FA). Still unable to connect to it, for some reason (see my other issue)

**Comment by landcraft** [comment]
I can confirm that, as already mentioned, gmail still works - you just need to enable 2FA and use an app password.
